backend/call-service/src/room.rs

```diff
@@ -0,0 +1,657 @@
+        track_remote::TrackRemote,
+    },
+};
+
+use crate::app::{AppStateMessage, RoomID};
+
+type UserID = Arc<String>;
+type TrackMap = Arc<DashMap<PacketIdentifier, TrackInfo>>;
+type ActiveUsersMap = Arc<DashMap<UserID, UserInfo>>;
+
+#[derive(Error, Debug, Clone, Copy)]
+pub enum Error {
+    #[error("User is not allowed in this room")]
+    UserNotAllowedInRoom,
+}
+
+#[derive(Debug, Deserialize, Serialize)]
+#[serde(tag = "type", rename_all = "lowercase")]
+pub enum WSFromUserMessage {
+    RTCOffer { offer: RTCSessionDescription },
+    RTCAnswer { answer: RTCSessionDescription },
+    RTCCandidate { candidate: RTCIceCandidateInit },
+    Disconected { user_id: UserID },
+    ConnectToRoom { room_id: String },
+}
+
+pub enum WSInnerUserMessage {
+    Message(Message),
+    Close,
+}
+
+#[derive(Debug, Clone)]
+pub enum TrackPacket {
+    UserPacket(Arc<UserPacket>),
+    //Opened(Arc<PacketIdentifier>),
+    Closed(Arc<PacketIdentifier>),
+    //UserLeft(Arc<UserID>),
+}
+
+#[derive(Debug)]
+pub enum RoomInfo {
+    PCStateChanged {
+        user_id: UserID,
+        state: RTCPeerConnectionState,
+    },
+    TrackCreated {
+        user_id: UserID,
+        track: Arc<TrackRemote>,
+    },
+    TrackRemoved {
+        identifier: Arc<PacketIdentifier>,
+    },
+    UserDisconected {
+        user_id: UserID,
+    },
+}
+
+#[derive(Debug, Hash, PartialEq, Eq, Clone)]
+pub struct PacketIdentifier {
+    sender: UserID,
+    track_id: String,
+}
+
+#[derive(Debug)]
+pub struct UserPacket {
+    identifier: Arc<PacketIdentifier>,
+    packet: Packet,
+}
+
+#[derive(Debug)]
+pub struct UserInfo {
+    user_channel_sender: mpsc::Sender<WSInnerUserMessage>,
+    //user_channel_receiver: mpsc::Receiver<WSFromUserMessage>,
+    connection: Arc<RTCPeerConnection>,
+    other_tracks: TrackMap,
+    user_tracks: Arc<DashMap<String, Arc<TrackRemote>>>,
+}
+
+impl UserInfo {
+    pub fn new(
+        user_channel_sender: mpsc::Sender<WSInnerUserMessage>,
+        //user_channel_receiver: mpsc::Receiver<WSFromUserMessage>,
+        connection: Arc<RTCPeerConnection>,
+    ) -> Self {
+        UserInfo {
+            user_channel_sender,
+            //user_channel_receiver,
+            connection,
+            other_tracks: Default::default(),
+            user_tracks: Default::default(),
+        }
+    }
+}
+
+#[derive(Debug)]
+pub struct TrackInfo {
+    track: Arc<TrackLocalStaticRTP>,
+    sender: Arc<RTCRtpSender>,
+}
+
+#[derive(Debug)]
+pub struct Room {
+    allowed_users: HashSet<String>,
+    active_users: ActiveUsersMap,
+    broadcast_sender: broadcast::Sender<TrackPacket>,
+    info_sender: mpsc::Sender<RoomInfo>,
+}
+
+impl Room {
+    pub async fn new(
+        allowed_users: Vec<String>,
+        sender: mpsc::Sender<AppStateMessage>,
+        id: RoomID,
+    ) -> Room {
+        let mut allowed_users_set = HashSet::default();
+        for user in allowed_users {
+            allowed_users_set.insert(user);
+        }
```

Comment:
```suggestion
        let mut allowed_users_set = HashSet::from_iterator(allowed_users);
```

`std::collections::HashSet` implements `FromIteratorTrait`, see [docs](https://doc.rust-lang.org/std/collections/struct.HashSet.html#impl-FromIterator%3CT%3E-for-HashSet%3CT,+S%3E)